skore/src/skore/sklearn/_plot/metrics/precision_recall_curve.py

```diff
@@ -910,3 +912,66 @@ def _compute_data_for_display(
             ml_task=ml_task,
             report_type=report_type,
         )
+
+    def frame(self, with_average_precision: bool = False) -> DataFrame:
+        """Get the data used to create the precision-recall curve plot.
+
+        Parameters
+        ----------
+        with_average_precision : bool, default=False
+            Whether to include the average precision column in the returned DataFrame.
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame containing the precision-recall curve data with columns
+            depending on the report type:
+
+            - estimator_name: Name of the estimator (when comparing estimators)
+            - split_index: Cross-validation fold ID (when doing cross-validation)
+            - label: Class label (for multiclass-classification)
+            - threshold: Decision threshold
+            - precision: Precision score at threshold
+            - recall: Recall score at threshold
+            - average_precision: average precision (when `with_average_precision=True`)
+
+        Examples
+        --------
+        >>> from sklearn.datasets import load_breast_cancer
+        >>> from sklearn.linear_model import LogisticRegression
+        >>> from skore import train_test_split, EstimatorReport
+        >>> X, y = load_breast_cancer(return_X_y=True)
+        >>> split_data = train_test_split(X=X, y=y, random_state=0, as_dict=True)
+        >>> classifier = LogisticRegression(max_iter=10_000)
+        >>> report = EstimatorReport(classifier, **split_data)
+        >>> display = report.metrics.precision_recall()
+        >>> df = display.frame()
+        """
+        if not with_average_precision:
+            total_frame = self.precision_recall
+        else:
+            total_frame = self.precision_recall.merge(
```

Comment:
Here, there is a bit of tricky pandas. Since we don't have index, pandas will use all common columns to make the merge (which is completely fine).

I think that we should add a comment to mention this since the code is not explicit and it comes to implementation details:

```suggestion
            # Merge the precision-recall curve and average precision
            # based on the common columns that are all columns but not the
            # one containing the statistics.
            total_frame = self.precision_recall.merge(
```

Reply:
It is also important to add a test that once a display is created, the dataframe `precision_recall` and `average_precision` have the exact same set of statistical columns. Having an assert will ensure that if we had a column in on dataframe in the future, the tests will fail and we will be aware that we need this correspondence for `merge to work.

Reply:
Added comments in https://github.com/probabl-ai/skore/commit/3493c8860cf2ee7c36e4db664a4154c0b2df67c6

Reply:
> It is also important to add a test that once a display is created, the dataframe `precision_recall` and `average_precision` have the exact same set of statistical columns. Having an assert will ensure that if we had a column in on dataframe in the future, the tests will fail and we will be aware that we need this correspondence for `merge to work.

1. You mean that I should add an assert to check that these two data frames have the same indexing columns (not the same statistical columns), right?
2. This assert should not be done inside of the `frame` method? Any suggestions on where to add such a test?

Reply:
> You mean that I should add an assert to check that these two data frames have the same indexing columns (not the same statistical columns), right?

Indeed.

> This assert should not be done inside of the frame method?

Yes it should be an additional test where we just construct a display by calling `report.metric.precision_recall()`.

```python
def test_precision_recall_constructor():
    report = ...  # create a report
    display = report.metrics.precision_recall()
    assert (
        display.precision_recall.select_dtypes(include=["category"]).columns.tolist() ==
        display.average_precision.select_dtypes(include=["category"].columns.tolist()
    )
```

In the same function you can then check more details regarding the expected value inside the `precision_recall` and `average_precision` columns.